aws-rds-globalcluster/src/main/java/software/amazon/rds/globalcluster/Translator.java

```diff
@@ -33,11 +35,19 @@ static CreateGlobalClusterRequest createGlobalClusterRequest(final ResourceModel
             .build();
   }
 
-  static ModifyGlobalClusterRequest modifyGlobalClusterRequest(final ResourceModel model) {
-    return ModifyGlobalClusterRequest.builder()
-            .globalClusterIdentifier(model.getGlobalClusterIdentifier())
-            .deletionProtection(model.getDeletionProtection())
-            .build();
+  static ModifyGlobalClusterRequest modifyGlobalClusterRequest(final ResourceModel previousModel, final ResourceModel desiredModel) {
+    ModifyGlobalClusterRequest.Builder builder = ModifyGlobalClusterRequest.builder()
+            .globalClusterIdentifier(desiredModel.getGlobalClusterIdentifier())
+            .deletionProtection(desiredModel.getDeletionProtection());
+
+    if (previousModel != null) {
+      if (!Objects.equals(previousModel.getEngineVersion(), desiredModel.getEngineVersion())) {
+        builder.engineVersion(desiredModel.getEngineVersion());
+        builder.allowMajorVersionUpgrade(true);
+      }
+    }
```

Comment:
I think you should incorporate `rollback` flag and if it is true and previous version is lower then current, then you should not include `EngineVersion` in the request, otherwise rollback will fail. Rollback is a revert update